src/map.cpp

```diff
@@ -116,7 +116,7 @@ map::map(int mapsize)
 {
     nulter = t_null;
     my_MAPSIZE = mapsize;
-    grid.resize( my_MAPSIZE * my_MAPSIZE, nullptr );
+    grid.resize( my_MAPSIZE * my_MAPSIZE * OVERMAP_LAYERS, nullptr );
```

Comment:
Yeah, I use the defines whenever possible.
Everything will break if OVERMAP_LAYERS definition ever becomes different from OVERMAP_DEPTH + OVERMAP_HEIGHT + 1, though. Though that would probably require some weird extra dimensions.